cargo-cyclonedx/src/generator.rs

```diff
@@ -182,60 +182,61 @@ impl SbomGenerator {
         let mut subcomponents: Vec<Component> = Vec::new();
         let mut subcomp_count: u32 = 0;
         for tgt in &package.targets {
-            // Ignore tests, benches, examples and build scripts.
-            // They are not part of the final build artifacts, which is what we are after.
-            if !(tgt.is_bench() || tgt.is_example() || tgt.is_test() || tgt.is_custom_build()) {
-                // classification
-                let cdx_type = if tgt.is_bin() {
-                    Classification::Application
-                } else if tgt.is_lib() {
-                    Classification::Library
-                } else {
-                    log::warn!("Target {} is neither a binary nor a library!", tgt.name);
-                    continue;
-                };
-
-                // bom_ref
-                let bom_ref = format!(
-                    "{} bin-target-{}",
-                    top_component.bom_ref.as_ref().unwrap(),
-                    subcomp_count
-                );
-                subcomp_count += 1;
-
-                // create the subcomponent
-                let mut subcomponent = Component::new(
-                    cdx_type,
-                    &tgt.name,
-                    &package.version.to_string(),
-                    Some(bom_ref),
-                );
-
-                // PURL subpaths are computed relative to the directory with the `Cargo.toml`
-                // *for this specific package*, not the workspace root.
-                // This is done because the tarball uploaded to crates.io only contains the package,
-                // not the workspace, so paths resolved relatively to the workspace root would not be valid.
-                //
-                // When using a git repo that contains a workspace, Cargo will automatically select
-                // the right package out of the workspace. Paths can then be resolved relatively to it.
-                // So the information we encode here is sufficient to idenfity the file in git too.
-                let package_dir = package
-                    .manifest_path
-                    .parent()
-                    .expect("manifest_path in `cargo metadata` output is not a file!");
-                if let Ok(relative_path) = tgt.src_path.strip_prefix(package_dir) {
-                    subcomponent.purl =
-                        get_purl(package, package, &self.workspace_root, Some(relative_path)).ok();
-                } else {
+            // Classification
```

Comment:
This PR seems to have dropped an important comment:

```
            // Ignore tests, benches, examples and build scripts.
            // They are not part of the final build artifacts, which is what we are after.
```

I'd like to have it re-added.

Reply:
Done in https://github.com/CycloneDX/cyclonedx-rust-cargo/pull/553/commits/5127338ae9472a0260969654ab41c3c470d13c3b